src/win32/posix_w32.c

```diff
@@ -516,6 +516,48 @@ int p_creat(const char *path, mode_t mode)
 	return p_open(path, O_WRONLY | O_CREAT | O_TRUNC, mode);
 }
 
+int p_fallocate(int fd, off_t offset, off_t len)
```

Comment:
Might make sense to have some tests in tests/core/posix.c, right?

Reply:
I added a test - I'm always a little nervous testing the POSIX stuff because it's easy to get caught up testing everybody else's implementation.  😀  But let's give it a shot.